Google.Cloud.AspNetCore/Google.AspNetCore.DataProtection.Kms/GoogleCloudDataProtectionBuilderExtensions.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright 2019 Google LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     https://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.AspNetCore.DataProtection.Kms;
+using Google.Cloud.Kms.V1;
+using Microsoft.AspNetCore.DataProtection.KeyManagement;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.DataProtection
+{
+    /// <summary>
+    /// Contains extension methods for modifying a <see cref="IDataProtectionBuilder"/> to work with Google KMS.
+    /// </summary>
+    public static class GoogleCloudDataProtectionBuilderExtensions
+    {
+        /// <summary>
+        /// Configures the data protection system to protect keys with specified key in Google Cloud KMS.
+        /// </summary>
+        /// <param name="builder">The data protection builder to configure.</param>
+        /// <param name="projectId">The Google Cloud project ID containing the KMS key ring.</param>
+        /// <param name="locationId">The location of the KMS key ring, e.g. "global".</param>
+        /// <param name="keyRingId">The ID of the KMS key ring.</param>
+        /// <param name="keyId">The ID of the key within the KMS key ring.</param>
+        /// <returns>The same builder, for chaining purposes.</returns>
+        public static IDataProtectionBuilder ProtectKeysWithGoogleKms(
+            this IDataProtectionBuilder builder,
+            string projectId,
+            string locationId,
+            string keyRingId,
+            string keyId) =>
+            ProtectKeysWithGoogleKms(builder, new CryptoKeyName(projectId, locationId, keyRingId, keyId));
+
+        /// <summary>
+        /// Configures the data protection system to protect keys with specified key in Google Cloud KMS.
+        /// </summary>
+        /// <param name="builder">The data protection builder to configure.</param>
+        /// <param name="keyName">The name of the KMS key to use.</param>
+        /// <returns>The same builder, for chaining purposes.</returns>
+        public static IDataProtectionBuilder ProtectKeysWithGoogleKms(
+            this IDataProtectionBuilder builder,
+            CryptoKeyName keyName) =>
```

Comment:
I'd like a convenient way to pass the keyname from `appsettings.json` into ProtectKeysWithGoogleKms().  Something like 
```json
    "KmsDataProtection": {
       "CryptoKeyName" : {
          "ProjectId": "my-project",
          "LocationId": "us-central1",
          "KeyRingId": "data-protector",
          "KeyId": "the-key"
        }
      }
```

That would require adding a class:
```cs
class KmsDataProtectionOptions
{
  public CryptoKeyName CryptoKeyName {get; set;}
}
```

And changing the constructor to accept an `IOptions<KmsDataProtectionOptions>`.

If we don't provide this, then many, my guess is the majority of users will implement this logic themselves.

Reply:
I'm not sure even that will quite do it, as `CryptoKeyName` itself is immutable. But yes, I'm certainly fine with us providing something like this. We may well want the more direct overload as well - but maybe not quite as many variations.